src/Sidekick.Infrastructure/PoeApi/Items/Modifiers/ModifierProvider.cs

```diff
@@ -205,28 +209,100 @@ private void ParseMods(List<Modifier> mods, List<ModifierPattern> patterns, stri
         {
             var unorderedMods = new List<Modifier>();
 
-            foreach (var data in patterns
-                .AsParallel()
-                .Where(x => x.Pattern != null && x.Pattern.IsMatch(text)))
+            foreach (var data in patterns)
             {
-                ParseMod(unorderedMods, text, data, data.Pattern.Match(text));
+                if (data.Pattern.IsMatch(text))
+                {
+                    ParseMod(unorderedMods, text, data, data.Pattern.Match(text));
+                }
+                else
+                {
+                    if (data.AlternateModifiers.Count > 0)
```

Comment:
I think this condition is unnecessary, if the count is 0, the foreach loop below will not execute either.